control/bdalg.py

```diff
@@ -325,15 +327,34 @@ def connect(sys, Q, inputv, outputv):
     >>> sysc = connect(sys, Q, [2], [1, 2])
 
     """
+    inputv, outputv, Q = np.asarray(inputv), np.asarray(outputv), np.asarray(Q)
+    # check indices
+    index_errors = (inputv - 1 > sys.inputs) | (inputv < 1)
+    if np.any(index_errors):  
+        raise IndexError( 
+            "inputv index %s out of bounds"%inputv[np.where(index_errors)])
+    index_errors = (outputv - 1 > sys.outputs) | (outputv < 1)
+    if np.any(index_errors):  
+        raise IndexError( 
+            "outputv index %s out of bounds"%outputv[np.where(index_errors)])
+    index_errors = (Q[:,0:1] - 1 > sys.inputs) | (Q[:,0:1] < 1)
+    if np.any(index_errors):  
+        raise IndexError( 
+            "Q input index %s out of bounds"%Q[np.where(index_errors)])
+    index_errors = (np.abs(Q[:,1:]) - 1 > sys.outputs)
+    if np.any(index_errors):  
+        raise IndexError( 
+            "Q output index %s out of bounds"%Q[np.where(index_errors)])
```

Comment:
- Style issue: no white space between format strings and argument. 
- %s is not the correct format. Indices are integers.

Reply:
fixed style issue.

re print: it to prints out an array of failing indices. is there a better way than just casting to a string like this?

Reply:
You're right. According to the [doc](https://docs.python.org/3/library/stdtypes.html#old-string-formatting), `%s` in old style formatting converts any argument using `str()`, which should always work for an array of integers.